front/src/applications/stdcmV2/components/StdcmInputVia.tsx

```diff
@@ -20,27 +24,68 @@ const StdcmInputVia = ({
     pathStep.stopFor ? `${ISO8601Duration2sec(pathStep.stopFor) / 60}` : ''
   );
 
+  const [showWarning, setShowWarning] = useState(false);
+
   const debounceUpdatePathStepStopTime = useMemo(
     () => debounce((value) => updatePathStepStopTime(value), 500),
     []
   );
 
+  useEffect(() => {
+    let newStopTime = pathStepStopTime;
+    let warning = false;
+
+    const isDriverSwitch = stopType === StdcmStopTypes.DRIVER_SWITCH;
+    const isPassageTime = stopType === StdcmStopTypes.PASSAGE_TIME;
+    const isServiceStop = stopType === StdcmStopTypes.SERVICE_STOP;
+
+    if (isPassageTime && pathStepStopTime !== '0') {
+      newStopTime = '0';
+    } else if (isDriverSwitch) {
+      const stopTimeValue = Number(pathStepStopTime);
+      if (stopTimeValue === 0) {
+        newStopTime = '3';
+      } else if (stopTimeValue < 3) {
+        warning = true;
+      }
```

Comment:
Maybe we can avoid one nested if with we config warning like that at the beginning

```suggestion
   const warning = isDriverSwitch && stopTimeValue < 3
```

Reply:
fixed